src/coreclr/jit/gentree.cpp

```diff
@@ -15460,7 +15522,7 @@ bool GenTree::DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarComm
             {
                 unsigned lclOffset = addrArgLcl->GetLclOffs();
 
-                if (lclOffset != 0)
+                if ((lclOffset != 0) || comp->lvaVarHiddenBufferStructArg(addrArgLcl->GetLclNum()))
```

Comment:
You should delete the `lvaVarHiddenBufferStructArg` condition here, this method takes a `width`, so any caller would either have to handle the call case explicitly, or is dealing with an `ASG(IND, x)`.

In the latter case, I am not actually convinced this is correct, it may lead some code to believe a full definition of a multi-def return buffer local (defined both by call and some `ASG`s) is partial while it is full (in other words, the `USEASG` flag will not match the `isEntire` value).

With that, `lvaVarHiddenBufferStructArg` itself should also be deleted, and the flag it checks for made `DEBUG`-only.

(I would actually prefer to delete it altogether and print the return buffer annotation as part of the argument message, as "return buffer" is really a property of the call argument, not the local (unlike "promoted", "address-exposed", etc), but up to you).

Reply:
Deleted `lvaVarHiddenBufferStructArg` because it doesn't need anymore. Regarding annotation, I find it useful to spot the "return buffer" in "Final variable assignment table" and then search disassembly code around that.